libkbfs/folder_branch_ops.go

```diff
@@ -1911,7 +1987,7 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		} else {
 			bid = fbo.bid
 		}
-		err := mdops.PutUnmerged(ctx, md, bid)
+		mdID, err = fbo.doPutWithCancellationCheck(ctx, lState, md, bid)
```

Comment:
Ah I see!

Damn it I just realize the checks done in `doPutWithCancellationCheck` after the `Put` doesn't make sense since if the `errPut` is `context.Canceled`, it means the context is canceled after all (even with the `EnterCriticalWithTimeout`). At that point, the `runUnlessCanceled` would have returned early and we still have  problems.